cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,981 @@
+
+The `fixed_width_column_wrapper` class should be used for constructing and initializing a column of any fixed-width element type, e.g., numeric types, timestamp types, boolean, etc.
+
+`fixed_width_column_wrapper` provides constructors that accept an iterator range to generate each element in the column.
+For nullable columns, an additional iterator can be provided indicating the validity of each element.
+There are also constructors that accept a `std::initializer_list<T>` for the column elements and optionally for the validity of each element.
+
+Example:
+```c++
+// Creates a non-nullable column of INT32 elements with 5 elements: {0, 1, 2, 3, 4}
+auto elements = make_counting_transform_iterator(0, [](auto i){return i;});
+fixed_width_column_wrapper<int32_t> w(elements, elements + 5);
+
+// Creates a nullable column of INT32 elements with 5 elements: {null, 1, null, 3, null}
+auto elements = make_counting_transform_iterator(0, [](auto i){return i;});
+auto validity = make_counting_transform_iterator(0, [](auto i){return i%2;})
+fixed_width_column_wrapper<int32_t> w(elements, elements + 5, validity);
+
+// Creates a non-nullable INT32 column with 4 elements: {1, 2, 3, 4}
+fixed_width_column_wrapper<int32_t> w{{1, 2, 3, 4}};
+
+// Creates a nullable INT32 column with 4 elements: {1, NULL, 3, NULL}
+fixed_width_column_wrapper<int32_t> w{ {1,2,3,4}, {1, 0, 1, 0}};
+```
+
+#### `strings_column_wrapper`
+
+The `strings_column_wrapper` class should be used for creating columns of strings. It provides constructors that accept an iterator range to generate each string in the column. 
+For nullable columns, an additional iterator can be provided indicating the validity of each string.
+There are also constructors that accept a `std::initializer_list<std::string>` for the column's strings and optionally for the validity of each element.
+
+Example:
+```c++
+// Creates a non-nullable STRING column with 7 string elements: 
+// {"", "this", "is", "a", "column", "of", "strings"}
+std::vector<std::string> strings{"", "this", "is", "a", "column", "of", "strings"};
+strings_column_wrapper s(strings.begin(), strings.end());
+
+// Creates a nullable STRING column with 7 string elements: 
+// {NULL, "this", NULL, "a", NULL, "of", NULL}
+std::vector<std::string> strings{"", "this", "is", "a", "column", "of", "strings"};
+auto validity = make_counting_transform_iterator(0, [](auto i){return i%2;});
+strings_column_wrapper s(strings.begin(), strings.end(), validity);
+
+// Creates a non-nullable STRING column with 7 string elements: 
+// {"", "this", "is", "a", "column", "of", "strings"}
+strings_column_wrapper s({"", "this", "is", "a", "column", "of", "strings"});
+
+// Creates a nullable STRING column with 7 string elements: 
+// {NULL, "this", NULL, "a", NULL, "of", NULL}
+auto validity = make_counting_transform_iterator(0, [](auto i){return i%2;});
+strings_column_wrapper s({"", "this", "is", "a", "column", "of", "strings"}, validity);
+```
+
+
```

Comment:
Should add a section on `fixed_point_column_wrapper`. @codereport

Reply:
Done.